src/main/java/com/checkmarx/flow/service/JiraService.java

```diff
@@ -255,16 +259,23 @@ private List<Issue> getIssues(ScanRequest request,String scannerFilter) {
         HashSet<String> fields = new HashSet<>();
         Collections.addAll(fields, "key", "project", "issuetype", "summary", LABEL_FIELD_TYPE, "created", "updated", "status");
 
-
-        SearchResult searchResults;
-        int totalResultsCount = MAX_RESULTS_ALLOWED;
-        SearchRestClient searchClient = this.client.getSearchClient();
-        //Retrieve JQL results through pagination (jira.max-jql-results per page -> default 50), don't allow less than 10.
-        int maxJqlResultsPerPage = Integer.max(10, jiraProperties.getMaxJqlResults());
-        for (int startAt = 0; startAt < totalResultsCount; startAt += maxJqlResultsPerPage) {
-            searchResults = searchClient.searchJql(jql, maxJqlResultsPerPage, startAt, fields).claim();
-            searchResults.getIssues().forEach(issues::add);
-            totalResultsCount = Integer.min(searchResults.getTotal(), MAX_RESULTS_ALLOWED);
+        //perform enhanced search using rest template when using JIRA cloud else use old search method
+        if(jiraProperties.getDeployType().equalsIgnoreCase("cloud")){
```

Comment:
As Cloud as most common case, we should keep cloud as default case.  
"if(StringUtils.isEmpty(jiraProperties.getDeployType()) ||       jiraProperties.getDeployType().equalsIgnoreCase("cloud"))"